server/auth/lib.py

```diff
@@ -135,14 +140,25 @@ def check_access(
                 description=f"{user.email} does not have access to jurisdiction {jurisdiction.id}"
             )
 
-    else:
-        assert user_type == UserType.AUDIT_BOARD
+    elif user_type == UserType.AUDIT_BOARD:
         assert audit_board
         if audit_board.id != user_key:
             raise Forbidden(
                 description=f"User does not have access to audit board {audit_board.id}"
             )
 
+    else:
+        assert user_type == UserType.TALLY_ENTRY
+        user = TallyEntryUser.query.filter_by(id=user_key).one()
+        if user.login_confirmed_at is None:
+            raise Unauthorized(
+                "Your jurisdiction manager must confirm your login code."
+            )
+        if jurisdiction and user.jurisdiction_id != jurisdiction.id:
+            raise Forbidden(
+                description=f"User does not have access to jurisdiction {jurisdiction.id}"
+            )
```

Comment:
We don't use `restrict_access` with tally entry users in this PR, but we implement and test it here, since it's a relevant part of the auth scheme. It will be used in coming PRs to authorize batch auditing endpoints.